.github/workflows/tools/update-issues/main.js

```diff
@@ -0,0 +1,223 @@
+          comment.createdAt >= issue.closedAt &&
+          body.includes('@shaka-bot') &&
+          (body.includes('reopen') || body.includes('re-open'))) {
+        core.notice(`Found reopen request for issue #${issue.number}`);
+        await issue.reopen();
+        break;
+      }
+    }
+  }
+}
+
+async function manageWaitingIssues(issue) {
+  // Find all waiting issues.
+  if (!issue.closed && issue.hasLabel(STATUS_WAITING)) {
+    const labelAgeInDays = await issue.getLabelAgeInDays(STATUS_WAITING);
+
+    // If an issue has been replied to, remove the waiting tag.
+    // Important: only load comments if prior filters pass!
+    // If we loaded them on every issue, we could exceed our query quota!
+    await issue.loadComments();
+
+    const latestNonTeamComment = issue.comments.find(c => !c.fromTeam);
+    if (latestNonTeamComment &&
+        latestNonTeamComment.ageInDays < labelAgeInDays) {
+      await issue.removeLabel(STATUS_WAITING);
+      return;
+    }
+
+    // If an issue has been in a waiting state for too long, close it as stale.
+    if (labelAgeInDays >= CLOSE_AFTER_WAITING_DAYS) {
+      await issue.postComment(CLOSE_STALE_TEXT);
+      await issue.close();
+    }
+  }
+}
+
+async function cleanUpIssueTags(issue) {
+  // If an issue with the waiting tag was closed, remove the tag.
+  if (issue.closed && issue.hasLabel(STATUS_WAITING)) {
+    await issue.removeLabel(STATUS_WAITING);
+  }
+}
+
+async function pingQuestions(issue) {
+  // If a question hasn't been responded to recently, ping it.
+  if (!issue.closed &&
+      issue.hasLabel(TYPE_QUESTION) &&
+      !issue.hasLabel(STATUS_WAITING)) {
+    // Important: only load comments if prior filters pass!
+    // If we loaded them on every issue, we could exceed our query quota!
+    await issue.loadComments();
+
+    // Most recent ones are first.
+    const lastComment = issue.comments[0];
+    if (lastComment &&
+        lastComment.fromTeam &&
+        // If the last comment was from the team, but not from the OP (in case
+        // the OP was a member of the team).
+        lastComment.author != issue.author &&
+        lastComment.ageInDays >= PING_INACTIVE_QUESTION_DAYS) {
+      await issue.postComment(`@${issue.author} ${PING_QUESTION_TEXT}`);
+      await issue.addLabel(STATUS_WAITING);
+    }
+  }
+}
+
+async function maintainMilestones(issue, nextMilestone, backlog) {
+  // Set or remove milestones based on type labels.
+  if (!issue.closed) {
+    if (issue.hasAnyLabel(LABELS_FOR_NEXT_MILESTONE)) {
+      if (!issue.milestone) {
+        await issue.setMilestone(nextMilestone);
+      }
+    } else if (issue.hasAnyLabel(LABELS_FOR_BACKLOG)) {
+      if (!issue.milestone) {
+        await issue.setMilestone(backlog);
+      }
+    } else {
+      if (issue.milestone) {
+        await issue.removeMilestone();
+      }
+    }
+  }
+}
+
+
+const ALL_ISSUE_TASKS = [
+  archiveOldIssues,
+  unarchiveIssues,
+  reopenIssues,
```

Comment:
Since the `reopenIssues` step happens after the `archiveOldIssues` step, does that mean that we might run into a situation where someone asks to re-open an issue at the very last moment and it archives instead?

Reply:
Good point.  The original tool may have has this problem as well, since it was based on a python decorator and no explicit ordering.  Fixed by moving reopenIssues to the top.